Content.Server/_Citadel/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,177 @@
+
+namespace Content.Server.Shipyard.Systems
+{
+
+    public sealed partial class ShipyardSystem : EntitySystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly PricingSystem _pricing = default!;
+        [Dependency] private readonly ShuttleSystem _shuttle = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly CargoSystem _cargo = default!;
+        [Dependency] private readonly MapLoaderSystem _map = default!;
+
+        public MapId? ShipyardMap { get; private set; }
+        private float _shuttleIndex;
+        private const float ShuttleSpawnBuffer = 1f;
+        private ISawmill _sawmill = default!;
+
+        public override void Initialize()
+        {
+            _sawmill = Logger.GetSawmill("shipyard");
+            SubscribeLocalEvent<BecomesStationComponent, ComponentStartup>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, BecomesStationComponent component, ComponentStartup args)
+        {
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            CleanupShipyard();
+        }
+
+        /// <summary>
+        /// Adds a ship to the shipyard, calculates its price, and attempts to ftl-dock it to the given station
+        /// </summary>
+        /// <param name="Station ID"></param>
+        /// <param name="Shuttle Path"></param>
+        public void PurchaseShuttle(EntityUid? stationUid, string shuttlePath)
+        {
+            if (!TryComp<StationDataComponent>(stationUid, out var stationData) || !TryComp<ShuttleComponent>(AddShuttle(shuttlePath), out var shuttle))
+                return;
+
+            var targetGrid = _station.GetLargestGrid(stationData);
+
+            if (targetGrid == null)
+                return;
+
+            var price = _pricing.AppraiseGrid(shuttle.Owner, null);
+
+            //can do FTLTravel later instead if we want to open that door
+            _shuttle.TryFTLDock(shuttle, targetGrid.Value);
+
+            _sawmill.Info($"Shuttle {shuttlePath} was purchased at {targetGrid} for {price} spacebucks");
+        }
+
+        /// <summary>
+        /// Loads a paused shuttle into the ShipyardMap from a file path
+        /// </summary>
+        /// <param name="Shuttle Path"></param>
+        /// <returns></returns>
+        private EntityUid? AddShuttle(string shuttlePath)
+        {
+            if (ShipyardMap == null)
+                return null;
+
+            //only dealing with a single grid at a time for now
+
+            var loadOptions = new MapLoadOptions()
+            {
+                Offset = (500f + _shuttleIndex, 0f)
+            };
+
+            var shuttle = _map.LoadGrid(ShipyardMap.Value, shuttlePath.ToString(), loadOptions);
```

Comment:
LoadGrid is deprecated. Use TryLoad and the appropriate MapLoad settings.

Reply:
TryLoad is going to accept all map files, including multi grid or even _no_ grid files, and it wont return the grid 

If you insist on not using loadgrid, I'm going to end up having to re-write the LoadGrid function inside of this method anyways